packages/composites/proof-of-reserves/src/utils/addressValidator.ts

```diff
@@ -0,0 +1,75 @@
+import { utils } from 'ethers'
+import { Logger } from '@chainlink/ea-bootstrap'
+
+type AddressObject = {
+  address: string
+  network?: string
+}
+
+export const validateAddresses = (indexer: string, addresses: AddressObject[]): AddressObject[] => {
+  const validatedAddresses: AddressObject[] = []
+  for (const addressObj of addresses) {
+    const { address, network } = addressObj
+    let validatedAddress: string | undefined
+    if (network) {
```

Comment:
Instead of checking if `network` is set and doing two switch-es here, I would prefer we do something like this:

```typescript
const indexerToNetwork = {
  'eth_balance': 'ethereum'
}

const validationNetwork = network || indexerToNetwork[indexer]
```

This would allow us to only do a single switch, so we don't duplicate this logic. Does this make sense?

Reply:
Yeah this is a much better way to do this in order to avoid duplicated logic.  I have implemented it as you suggested.